Freethreading documentation (#6298)
cython/cython: docs/src/userguide/freethreading.rst

*************

> Freethreading

Reviewer: Let's use the term like it's used in the Python docs.

```suggestion
Free threading
```

*************

> The Freethreading (sometimes known as "nogil") build of Python is an

Reviewer: Same comment as above. Ditto in other places.

```suggestion
The free-threaded (sometimes known as "nogil") build of Python is an
```

experimental mode available from Python 3.13 onwards. It aims to

> remove the "Global Interpreter Lock" and allow multiple Python threads to

Reviewer: ```suggestion
disable the "Global Interpreter Lock" and allow multiple Python threads to
```
